### ISSUE TITLE
Confusion about instruction task

### ISSUE THREAD

**Issue by mitchelldehaven** [tatsu-lab/stanford_alpaca#299]
I'm trying to do some instruction fine-tuning and using your dataset as a reference for implementation. However, one portion I don't quite understand, but I'm assuming it is not a bug since I've seen this implemented this way in multiple places, is how the input ids and labels are generated for the instruction tasks. The input ids and labels are not offset, it just masks the prompt inputs. For example, you might have an input and label as below:

```
input_ids = [1, 1951, 31, 3642, 9342, 1002, 2] # assume token_ids 1, 1951, 31 are part of the prompt
labels = [-100, -100, -100, 3642, 9342, 1002, 2]
```

In this example, at index 3, the input token id is 3642 and the target at the same time step is 3642. This doesn't really make sense to me, it seems like a trivial task, because we are simply asking the model to predict the token we are currently inputting, rather than the next token in the sequence (like it does during pretraining). Can someone help me understand why this is the case?

It looks like #290 is pointing out something similar, however it was dismissed, stating that the attention mask will address the problem. How can the attention mask address the fact that the input at index 3 is telling the model what the target at index 3 is, they are the same token id?

**Comment by mitchelldehaven** [comment]
Closing this as I found out what I wasn't understanding correctly. The CasualLM class from Transformers automatically offset these values in the forward pass, so the input to the model class above is correct, since it is then shifted correctly in the loss computation.
